packages/flutter_tools/lib/src/cache.dart

```diff
@@ -535,12 +538,19 @@ abstract class EngineCachedArtifact extends CachedArtifact {
 
       _makeFilesExecutable(dir);
 
+      // TODO(jonahwilliams): clean this logic up to better handle different framework names.
```

Comment:
How about something like:
```
final List<String> frameworkNames = ['Flutter', 'FlutterMacOS'];
for (frameworkName in frameworkNames) {
  final File frameworkZip = fs.file(fs.path.join(dir.path, '$frameworkName.framework.zip'));
  ...
}
```
rather than duplicating the code.

Reply:
Done